actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -145,19 +177,137 @@ def render_to_body(options)
     end
 
     def _render_to_body_with_renderer(options)
-      _renderers.each do |name|
-        if options.key?(name)
-          _process_options(options)
-          method_name = Renderers._render_with_renderer_method_name(name)
-          return send(method_name, options.delete(name), options)
+      _renderers.each do |renderer_name|
+        next unless options.key?(renderer_name)
```

Comment:
I wonder if there's a particular reason it doesn't check `formats` here, so that `render @model` with `formats: [:json]` would behave like `render json: @model` does now. Maybe that's only relevant in ActionController::MimeResponds::ClassMethods.`respond_to` blocks.